### PR TITLE
[WB-7274]  Add support raw ip addresses for launch

### PR DESCRIPTION
**Pull request by KyleGoyette**
[Fixes WB-7274](https://wandb.atlassian.net/browse/WB-7274)

Description
-----------

- Adds logging lines
- adds support for raw ip addresses like local

Testing
-------

How was this PR tested?

Release Notes
-------------

Below, please enter user-facing release notes as one or more bullet points.
If your change is not user-visible, write `NO RELEASE NOTES` instead, with no bullet points.

------------- BEGIN RELEASE NOTES ------------------
NO RELEASE NOTES

------------- END RELEASE NOTES --------------------

### CODE REVIEW

File: wandb/sdk/launch/_project_spec.py [hunk 1]
```diff
@@ -148,10 +159,13 @@ def get_entry_point(self, entry_point: str) -> "EntryPoint":
     def _fetch_project_local(self, internal_api: Api) -> None:
         """Fetch a project into a local directory, returning the path to the local project directory."""
         assert self.source != LaunchSource.LOCAL
+        _logger.info("Fetching project locally...")
         if utils._is_wandb_uri(self.uri):
+            _logger.info("Fetching run info...")
```
Comment (stephchen) [inline wandb/sdk/launch/_project_spec.py:164]: this and line 168 seem to be duplicates?

File: wandb/sdk/launch/docker.py [hunk 2]
```diff
@@ -231,7 +239,7 @@ def get_docker_command(
         _, _, port = _, _, port = api.settings("base_url").split(":")
         base_url = "http://host.docker.internal:{}".format(port)
     elif _is_wandb_dev_uri(api.settings("base_url")):
-        base_url = "http://host.docker.internal:9002"
+        base_url = "http://host.docker.internal:9003"
```
Comment (stephchen) [inline wandb/sdk/launch/docker.py:242]: out of curiosity what's the reason for this change?

File: wandb/sdk/launch/utils.py [hunk 3]
```diff
@@ -163,6 +165,19 @@ def parse_wandb_uri(uri: str) -> Tuple[str, str, str]:
     return entity, project, name
 
 
+def is_bare_wandb_uri(uri: str) -> bool:
+    """Checks if the uri is of the format /entity/project/runs/run_name"""
```
Comment (stephchen) [inline wandb/sdk/launch/utils.py:169]: when do we expect this format? especially with the `runs` in the middle which would reasonably only appear if you're pasting from an actual url and cutting off the front

**Review by stephchen — COMMENTED** [review]
in general log lines look great, although i wonder if this many lines (especially starting and ending lines for operations that shouldn't take any time, eg "clearing/cleared parameter collisions") is necessary/if it gets noisy

File: wandb/sdk/launch/docker.py (hunk 2 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/docker.py:242]: Recently the dev port changed from 9002 to 9003, this makes launch work with api.wandb.test

File: wandb/sdk/launch/utils.py (hunk 3 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/utils.py:169]: Unfortunately some local installations seem to send the uri in the spec in this format. Filed a ticket for it: https://wandb.atlassian.net/browse/WB-7594

File: wandb/sdk/launch/_project_spec.py (hunk 1 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/_project_spec.py:164]: Ah good catch.

**Comment by KyleGoyette** [comment]
I'll tone the logging down a bit. This was extensive to minimize testing for a user who was trying it out.

**Review by stephchen — APPROVED** [review]
